cloudflare/resource_cloudflare_access_policy.go

```diff
@@ -66,8 +66,70 @@ func resourceCloudflareAccessPolicy() *schema.Resource {
 				Required: true,
 				Elem:     AccessGroupOptionSchemaElement,
 			},
+			"purpose_justification_required": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"purpose_justification_prompt": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"purpose_justification_required"},
+			},
+			"approval_group": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     AccessPolicyApprovalGroupElement,
+			},
+		},
+	}
+}
+
+var AccessPolicyApprovalGroupElement = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"email_list_uuid": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"email_addresses": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+		"approvals_needed": {
+			Type:         schema.TypeInt,
+			Required:     true,
+			ValidateFunc: validation.IntAtLeast(0),
 		},
+	},
+}
+
+func APIAccessPolicyApprovalGroupToSchema(approvalGroup cloudflare.AccessApprovalGroup) map[string]interface{} {
+	data := make(map[string]interface{})
+	data["approvals_needed"] = approvalGroup.ApprovalsNeeded
+
+	if approvalGroup.EmailAddresses != nil {
+		data["email_addresses"] = approvalGroup.EmailAddresses
+	}
+
+	if approvalGroup.EmailListUuid != "" {
+		data["email_list_uuid"] = approvalGroup.EmailListUuid
 	}
+	return data
+}
+
+func SchemaAccessPolicyApprovalGroupToApi(data map[string]interface{}) cloudflare.AccessApprovalGroup {
```

Comment:
similar to https://github.com/cloudflare/terraform-provider-cloudflare/pull/1199/files#r709540696

```suggestion
func schemaAccessPolicyApprovalGroupToApi(data map[string]interface{}) cloudflare.AccessApprovalGroup {
```